src/intrinsics/ecma262/Object.js

```diff
@@ -140,7 +140,13 @@ export default function(realm: Realm): NativeFunctionValue {
     }
 
     // 5. Return to.
-    if (to_must_be_partial) to.makePartial();
+    if (to_must_be_partial) {
+      to.makePartial();
+      // at this point, we know the target is partial and started with
+      // no keys, otherwise the above FatalError() would have been thrown
+      // so we can safely make it simple
+      to.makeSimple();
```

Comment:
I think this makes sense. But I would feel better if the assumptions we're making (e.g. that FatalError *would* have been thrown by now) were somehow written in code. [The line says](https://github.com/trueadm/prepack/blob/729c81f832b3c073f5b768781b9ac2f72ca054fc/src/intrinsics/ecma262/Object.js#L86) we don't "yet" support this. What if we later add support? Then the assumption in this code branch would be invalid.

Reply:
I've add some code that follows the assumptions. If they were changed and the FatalError wasn't thrown, then the object wouldn't be made simple.